distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -2500,6 +2506,78 @@ private void checkExpectations(CometDClient cometDClient, Integer numNotificatio
 
     }
 
+    /**
+     * Confirms that the "download resource to local cache" action's url can be
+     * found in the CometDClient messages and matches the expected url.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void findMetacardActionUrlUsingCometDClient() throws Exception {
+
+        String metacardId = ingestXmlWithProduct(String.format("%s.txt", testName.getMethodName()));
+
+        metacardsToDelete.add(metacardId);
+
+        final String RESPONSE_CHANNEL = "/0193d9e7f9ed4f8f8bd02103143c41d6";
+
```

Comment:
Probably need to do some testing here.  I don't think we need to subscribe to all of those channels for this test to work.